fedimint-core/src/bitcoin_migration.rs

```diff
@@ -0,0 +1,196 @@
+use std::str::FromStr;
+
+use bitcoin30::hashes::Hash as Bitcoin30Hash;
+use bitcoin_hashes::hex::FromHex;
+use bitcoin_hashes::Hash as Bitcoin29Hash;
+
+pub fn bitcoin29_to_bitcoin30_public_key(pk: bitcoin::PublicKey) -> bitcoin30::PublicKey {
+    bitcoin30::PublicKey::from_slice(&pk.to_bytes()).unwrap()
+}
+
+pub fn bitcoin29_to_bitcoin30_network(network: bitcoin::Network) -> bitcoin30::Network {
+    match network {
+        bitcoin::Network::Bitcoin => bitcoin30::Network::Bitcoin,
+        bitcoin::Network::Testnet => bitcoin30::Network::Testnet,
+        bitcoin::Network::Signet => bitcoin30::Network::Signet,
+        bitcoin::Network::Regtest => bitcoin30::Network::Regtest,
+    }
+}
+
+pub fn bitcoin30_to_bitcoin29_script(script: bitcoin30::ScriptBuf) -> bitcoin::Script {
+    bitcoin::Script::from_hex(&script.to_hex_string()).unwrap()
```

Comment:
We shouldn't use `unwrap` here, `expect` with a descriptive error message is better. 

I'm not sure how we should handle errors here though, should we assume that every `bitcoin30::ScriptBuf` is a valid `bitcoin::Script`?

Reply:
`Script`/`ScriptBuf` doesn't do any semantic validation. The error comes from the hex decoding, which isn't necessary since we can use the [`from/to_bytes`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/script/struct.ScriptBuf.html#method.from_bytes) methods instead

Reply:
Can you use `from/to_bytes` like @elsirion mentioned here?

Reply:
Done.